WordPress/gutenberg
# Fix font library modal dialog translatable strings

Fixes https://github.com/WordPress/gutenberg/issues/58201

## What?

A couple translatable strings are not translatable as they don't follow the WordPress localization best practices and are unnecessarily comples.

## Why?

Following the WordPress localization best practices makes sure strings are translatable to all languages. For example:
- Always provide full sentences instead of split sentences.
- Avoid concatenation.
- Use `sprintg` and placeholder appropriately.
- The liste item separator `, ` must be translatable as well.

## How?

- Simplifies the strings and use `sprintf` where appropriate.
- Avodis concatenation.
- ~Removes a `__( 'and' )` translatable string anc concatenation.~
- ~Introduces a `l10n` object in the `editor/utils` package. For now, it only contains a `listItemSeparator` string. In the future this could be expanded with more strings, reshaped, or even moved to a separate package if desired. Pleaae let me know if the current approach is acceptable Cc @youknowriad @SergeyBiryukov~
- Update: simplifies the allowed font types string.


Note: in Core, the list item separator comes from  `WP_Locale->get_list_item_separator()`. It is in a centralized location and can be reused across the codebase. I'd think something similar is worth introducing also in Gutenberg.

## Testing Instructions




- Go to Site editor > Styles > Typography > Manage fonts.
- In the Fonts modal dialog, go to the Google fonts tab.
- Observe the indication of the font variants is correct e.g. '1 variant', '2 variants', etc.
- Go to the Upload tab.
- Observe the descriptive text of the allowed font types is unchanged.

### Testing Instructions for Keyboard


## Screenshots or screencast
